addons/survey/views/survey_templates.xml

```diff
@@ -351,21 +431,36 @@
                 </tr>
             </thead>
             <tbody>
+                <t t-set="letters" t-value="'ABCDEFGHIJKLMNOPQRSTUVWXY'"/>
+                <t t-set="item_idx" t-value="0"/>
                 <tr t-foreach="question.matrix_row_ids" t-as="row_label">
                     <th><span t-field="row_label.value" /></th>
-                    <td t-foreach="question.suggested_answer_ids" t-as="col_label">
+                    <t t-foreach="question.suggested_answer_ids" t-as="col_label">
                         <t t-set="answer" t-value="answer_lines.filtered(lambda line: line.suggested_answer_id == col_label and line.matrix_row_id == row_label)"/>
-                        <input t-att-type="'checkbox' if question.matrix_subtype == 'multiple' else 'radio'"
-                               t-att-name="'%s_%s' % (question.id, row_label.id)" t-att-value='col_label.id'
-                               t-att-checked="'checked' if answer else None"
-                               t-att-data-row-id="row_label.id"/>
-                    </td>
+                        <td t-att-class="'o_survey_matrix_btn' + (' o_survey_selected ' if answer else '')">
+                            <input t-att-type="'checkbox' if question.matrix_subtype == 'multiple' else 'radio'"
+                                   t-att-name="'%s_%s' % (question.id, row_label.id)" t-att-value='col_label.id'
+                                   t-att-checked="'checked' if answer else None"
+                                   t-att-data-row-id="row_label.id"
+                                   class="o_survey_form_choice_item d-none"/>
+                            <div t-att-class="'o_survey_matrix_empty_checkbox' + (' o_survey_radio_btn ' if question.matrix_subtype != 'multiple' else '')"/>
+                            <i t-if="question.matrix_subtype == 'multiple'" class="fa fa-check-square"></i>
+                            <i t-else="" class="fa fa-check-circle"></i>
+                            <div t-if="useKeySelection"
+                                 t-att-class="'o_survey_choice_key float-right font-weight-bold' + (' o_survey_radio_btn ' if question.matrix_subtype != 'multiple' else '')">
+                                 <span class="o_survey_key"><span>Key</span></span>
+                                 <span t-esc="letters[item_idx]"/>
+                            </div>
```

Comment:
Maybe that key assignation thing could be factorized a bit, but eeeeeeh not sure.